### PR TITLE
support tls connection

### PR DESCRIPTION
**Pull request by id**

### CODE REVIEW

**Comment by id** [comment]
Fixes #76

File: src/brod_sock.erl [hunk 1]
```diff
@@ -139,28 +150,44 @@ debug(Pid, File) when is_list(File) ->
...
+                [] ->
+                  State0#state{mod = gen_tcp, sock = Sock};
+                SslOpts ->
+                  error_logger:info_msg("Trying to establish ssl connection with options: ~p\n", [SslOpts]),
+                  case ssl:connect(Sock, SslOpts, Timeout) of
+                    {ok, SslSock} ->
+                      State0#state{mod = ssl, sock = SslSock};
+                    {error, Reason} ->
+                      exit({ssl_negotiation_failure, Reason,
+                            erlang:get_stacktrace()})
```
Comment (zmstone) [inline src/brod_sock.erl]: why stacktrace here ?

Comment (id) [reply src/brod_sock.erl]: to get more detailed info in crash log

**Comment by zmstone** [comment]
looks good

**Pull request merged**